src/pages/Wallet/AddressDetailsPage.tsx

```diff
@@ -0,0 +1,157 @@
+import DataList, { DataListCell, DataListRow } from '../../components/DataList'
+import Label from '../../components/Label'
+import { MainContent, PageTitleRow } from '../../components/PageComponents/PageContainers'
+import { PageH1, PageH2 } from '../../components/PageComponents/PageHeadings'
+import Table, { TableProps } from '../../components/Table'
+import { useAddressesContext } from '../../contexts/addresses'
+import AddressOptionsModal from './AddressOptionsModal'
+
+const transactionsTableHeaders: TableProps['headers'] = [
+  { title: 'Direction' },
+  { title: 'Timestamp' },
+  { title: 'Address(es)' },
+  { title: 'Amount', align: 'end' }
+]
+
+const AddressDetailsPage = () => {
+  const [isAddressOptionsModalOpen, setIsAddressOptionsModalOpen] = useState(false)
+  const { addressesState } = useAddressesContext()
+  const { address } = useParams<{ address: string }>()
+  const addressData = addressesState.get(address)
+  const history = useHistory()
+
+  if (!addressData) return null
+
+  return (
+    <MainContent>
+      <PageTitleRow>
+        <Title>
+          <ArrowLeftStyled onClick={() => history.goBack()} />
+          <PageH1Styled>
+            Address details {addressData.settings.isMain && <MainAddress>Main address</MainAddress>}
+          </PageH1Styled>
+          {addressData.settings.label && (
+            <LabelStyled color={addressData.settings.color}>{addressData.settings.label}</LabelStyled>
+          )}
+          <OptionsButton
+            transparent
+            squared
+            onClick={() => setIsAddressOptionsModalOpen(true)}
+            aria-label="Address options"
+          >
+            <SettingsIcon />
+          </OptionsButton>
+        </Title>
+      </PageTitleRow>
+      <DataList>
+        <DataListRow>
+          <DataListCell>Address</DataListCell>
+          <DataListCell>
+            {address}
+            <ClipboardButton textToCopy={address} />
+            <QRCodeButton textToEncode={address} />
+            <OpenInExplorerButton address={address} />
+          </DataListCell>
+        </DataListRow>
+        <DataListRow>
+          <DataListCell>Label</DataListCell>
+          <DataListCell>
+            {addressData.settings.label ? (
+              <Label color={addressData.settings.color}>{addressData.settings.label}</Label>
+            ) : (
+              '-'
+            )}
+          </DataListCell>
+        </DataListRow>
+        <DataListRow>
+          <DataListCell>Number of transactions</DataListCell>
+          <DataListCell>-</DataListCell>
+        </DataListRow>
+        <DataListRow>
+          <DataListCell>ALPH balance</DataListCell>
+          <DataListCell>
+            {addressData.details?.balance ? (
+              <AmountStyled value={BigInt(addressData.details.balance)} fadeDecimals />
+            ) : (
+              '-'
+            )}
+          </DataListCell>
+        </DataListRow>
```

Comment:
Let's directly add the locked balance - as in the explorer - except that I would maybe display first the locked balance, and then the total balance:

- Locked ALPH balance (if any)
- Total ALPH balance